hub/core/chunk_engine.py

```diff
@@ -380,6 +403,39 @@ def update(self, index: Index, samples: Union[Sequence[SampleValue], SampleValue
             chunks_nbytes_after_updates, self.min_chunk_size, self.max_chunk_size
         )
 
+    def _update_samples_subslice(self, index: Index, incoming_samples: np.ndarray):
+        """Update the samples at `index` (must be a subslice index) with incoming samples.
+
+        Note:
+            This method requires the incoming samples' shapes to be exactly the same as the `index` subslice.
+            For full sample updates (allowing new shapes), use `_update_samples`.
+        """
+
+        index_shape = index.shape
+        if index_shape[0] != 1:
+            raise MultiSampleSubsliceUpdateError(index_shape)
+
+        # squeeze 1s away
+        squeezed_index_shape = tuple([dim for dim in index_shape if dim != 1])
```

Comment:
why do we squeeze all ones? I think we only need to squeeze ones from the very first dim